heron/stmgr/tests/cpp/server/stateful-restorer_unittest.cpp

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <limits>
+#include <map>
+#include <vector>
+#include <sstream>
+#include <thread>
+#include <string>
+#include <unordered_set>
+#include <unordered_map>
+#include "gtest/gtest.h"
+#include "glog/logging.h"
+#include "proto/messages.h"
+#include "basics/basics.h"
+#include "errors/errors.h"
+#include "threads/threads.h"
+#include "network/network.h"
+#include "basics/modinit.h"
+#include "errors/modinit.h"
+#include "threads/modinit.h"
+#include "network/modinit.h"
+#include "config/topology-config-vars.h"
+#include "config/topology-config-helper.h"
+#include "config/physical-plan-helper.h"
+#include "config/heron-internals-config-reader.h"
+#include "metrics/metrics-mgr-st.h"
+#include "manager/stateful-restorer.h"
+#include "server/dummy_ckptmgr_client.h"
+#include "server/dummy_tuple_cache.h"
+#include "server/dummy_stmgr_clientmgr.h"
+#include "server/dummy_stmgr_server.h"
+
+const sp_string SPOUT_NAME = "spout";
+const sp_string BOLT_NAME = "bolt";
+const sp_string STREAM_NAME = "stream";
+const sp_string CONTAINER_INDEX = "0";
+const sp_string STMGR_NAME = "stmgr";
+const sp_string LOCALHOST = "127.0.0.1";
+
+sp_string heron_internals_config_filename =
+    "../../../../../../../../heron/config/heron_internals.yaml";
+
+// Generate a dummy topology
+static heron::proto::api::Topology* GenerateDummyTopology(
```

Comment:
This is part of a bigger cleanup that touches other tests in the directory. Will send that as part of a cleanup in another pr